src/core/tsi/alts_transport_security.cc

```diff
@@ -20,44 +20,46 @@
 
 #include "src/core/tsi/alts_transport_security.h"
 
-#include <string.h>
+#include <new>
 
-static alts_shared_resource g_alts_resource;
+static alts_shared_resource *g_alts_resource;
 
 alts_shared_resource* alts_get_shared_resource(void) {
-  return &g_alts_resource;
+  return g_alts_resource;
 }
 
 static void grpc_tsi_alts_wait_for_cq_drain() {
-  gpr_mu_lock(&g_alts_resource.mu);
-  while (!g_alts_resource.is_cq_drained) {
-    gpr_cv_wait(&g_alts_resource.cv, &g_alts_resource.mu,
+  gpr_mu_lock(&g_alts_resource->mu);
+  while (!g_alts_resource->is_cq_drained) {
+    gpr_cv_wait(&g_alts_resource->cv, &g_alts_resource->mu,
                 gpr_inf_future(GPR_CLOCK_REALTIME));
   }
-  gpr_mu_unlock(&g_alts_resource.mu);
+  gpr_mu_unlock(&g_alts_resource->mu);
 }
 
 void grpc_tsi_alts_signal_for_cq_destroy() {
-  gpr_mu_lock(&g_alts_resource.mu);
-  g_alts_resource.is_cq_drained = true;
-  gpr_cv_signal(&g_alts_resource.cv);
-  gpr_mu_unlock(&g_alts_resource.mu);
+  gpr_mu_lock(&g_alts_resource->mu);
+  g_alts_resource->is_cq_drained = true;
+  gpr_cv_signal(&g_alts_resource->cv);
+  gpr_mu_unlock(&g_alts_resource->mu);
 }
 
 void grpc_tsi_alts_init() {
-  memset(&g_alts_resource, 0, sizeof(alts_shared_resource));
-  gpr_mu_init(&g_alts_resource.mu);
-  gpr_cv_init(&g_alts_resource.cv);
+  g_alts_resource = new alts_shared_resource;
```

Comment:
We are actually not able to use new/delete. Could you use gpr_malloc and gpr_free instead?

That should clear up the build errors

Reply:
(or gpr_zalloc I guess, since it looks like this expects to be zero-ed out)